htlcswitch/switch.go

```diff
@@ -493,14 +500,11 @@ func (s *Switch) GetAttemptResult(attemptID uint64, paymentHash lntypes.Hash,
 	// Since the attempt was known, we can start a goroutine that can
 	// extract the result when it is available, and pass it on to the
 	// caller.
-	s.wg.Add(1)
-	go func() {
-		defer s.wg.Done()
-
+	err = s.gm.Go(func(ctx context.Context) {
 		var n *networkResult
 		select {
 		case n = <-nChan:
-		case <-s.quit:
+		case <-ctx.Done():
```

Comment:
this will never happen cause we pass it a fresh `context.Background()` on instantiation. 

This makes me think that the `Go(func(ctx context.Context))` signature should actually be: `Go(func(ctx context.Context, quit chan struct{}))` where `quit` is a channel that is closed when the goroutine manager is exiting. That way we can handle that case inside here. How we handle the exiting of the goroutine manager is thus more explicit.

Reply:
the reason I'm pointing this out is cause we now have 2 places calling `close` on the `resultChan` which is a red flag to me (due to the potential of a panic). It takes introspection of the `Go` call to see that these two can never be called simultaneously, but I dont think a caller should need to depend on that the internals of the helper

Reply:
> this will never happen cause we pass it a fresh context.Background() on instantiation.

ah ok I see we do create a child context off of it and call cancel on that if GoroutingManager is stopped. 

I think we should still unify this though: explicit context passed in and an explicit quit channel called when GotroutineManager is Stopped. Ie, dont use the same derived context for 2 things. That way we dont have to handle the ErrStopping in 2 places as is done here.

Reply:
@ellemouton and me discussed the interface of `GoroutineManager.Go` method.
We are leaning towards this:
```go
Go(func(ctx context.Context)) bool

ok := Go(...)
if !ok {
  // Failed to launch a goroutine.
}
```

The quit channel is not needed, since the ctx is a child context which is cancelled upon goroutine manager stop.